util.go

```diff
@@ -17,26 +30,42 @@ func generateSigner() (ssh.Signer, error) {
 }
 
 func parsePtyRequest(s []byte) (pty Pty, ok bool) {
-	term, s, ok := parseString(s)
-	if !ok {
-		return
-	}
-	width32, s, ok := parseUint32(s)
-	if !ok {
-		return
+	reqMsg := &ptyRequestMsg{}
+	err := ssh.Unmarshal(s, reqMsg)
+	if err != nil {
+		return Pty{}, false
 	}
-	height32, _, ok := parseUint32(s)
-	if !ok {
-		return
+
+	modes := []byte(reqMsg.Modelist)
+
+	mode := struct {
+		Key uint8
+		Val uint32
+	}{}
+
+	TerminalModes := make(ssh.TerminalModes, 0)
+	for {
+		if len(modes) < 1 || modes[0] == ttyOPEND || len(modes) < 5 {
+			break
+		}
+		b := modes[:5]
+		err = ssh.Unmarshal(b, &mode)
+		if err != nil {
+			return Pty{}, false
+		}
+		TerminalModes[mode.Key] = mode.Val
+		modes = modes[6:]
 	}
+
 	pty = Pty{
-		Term: term,
+		Term: reqMsg.Term,
 		Window: Window{
-			Width:  int(width32),
-			Height: int(height32),
+			Width:  int(reqMsg.Columns),
+			Height: int(reqMsg.Rows),
 		},
+		TerminalModes: TerminalModes,
```

Comment:
This should be enough for now, but we may want to have an example or a convenience function that will handle this properly. Setting TerminalModes is not fun. This is just an example: https://github.com/golang/crypto/blob/master/ssh/terminal/util.go#L37

Reply:
Did you mean like this ?
```golang
func ApplyTerminalModes(fd int, terminalModes ssh.TerminalModes) bool {
	termios, err := unix.IoctlGetTermios(fd, unix.TCGETS)
	fmt.Println(termios, err)
	for opcode, val := range terminalModes {
		switch opcode {
		case ssh.TTY_OP_ISPEED:
			termios.TTY_OP_OSPEED = val
		case ssh.TTY_OP_OSPEED:
			termios.TTY_OP_OSPEED = val
		
		// case char
		case ssh.VINTR:
			termios.Cc[ssh.VINTR] = val
		case ssh.VQUIT:
			termios.Cc[ssh.VQUIT] = val
		//...
		//case mode
		case ssh.ECHO:
			if val {
				termios.Lflag |= ssh.ECHO
			} else {
				termios.Lflag &= ~ssh.ECHO
			}

		default
			// Others modes ...
		}

	}

	if err := unix.IoctlSetTermios(fd, unix.TCSETS, termios); err != nil {
		return nil, err
	}
	return true
}
```

This is the reference code https://github.com/openssh/openssh-portable/blob/1a7217ac063e48cf0082895aeee81ed2b8a57191/ttymodes.c#L397, the syntax may not be correct.